api/helpers/game-states/create-socket-event.js

```diff
@@ -29,7 +29,10 @@ module.exports = {
 
       const countPasses = (function () {
         let numPasses = 0;
-        for (const gameState of game.gameStates.slice(-3)) {
+        const currentIndex = game.gameStates.findIndex(gs => gs.id === gameState.id);
+        const threeUpToCurrent = game.gameStates.slice(Math.max(0, currentIndex - 2), currentIndex + 1);
+
+        for (const gameState of threeUpToCurrent) {
```

Comment:
This was the crux of the fix. Previously, we always looked at the 'latest three' game states to count passes, but for replays, we need to look at the 'three states leading up to the one we're looking at'. Doing _that_ requires the ability to identify which gamestate is which, so I've added the id from the gamestate row onto the game state obj via `pack-gamestate`